elements/rh-badge/rh-badge.ts

```diff
@@ -0,0 +1,28 @@
+import { customElement, property } from 'lit/decorators.js';
+
+import { BaseBadge } from '@patternfly/pfe-badge/BaseBadge.js';
+
+import styles from './rh-badge.css';
+
+/**
+ * A badge is used to annotate other information with numerical content.
+ */
+
+@customElement('rh-badge')
+export class RhBadge extends BaseBadge {
+  static readonly version = '{{version}}';
+
+  static readonly styles = [...BaseBadge.styles, styles];
+
+  /**
+   * Denotes the state-of-affairs this badge represents
+   * Options include read and unread
+   */
+  @property({ reflect: true }) state?: 'info'|'success'|'moderate'|'important'|'critical';
```

Comment:
this will fail currently. blocked by https://github.com/patternfly/patternfly-elements/pull/2259

When two subclasses need to declare different properties with the same name, the property needs to have an abstract declaration on the baseclass with a type wide enough to encompass both cases.

Reply:
I pushed some code which should fix this once we release pfe-badge and update the version here